### ISSUE TITLE
ActorOfAsTestActorRef does not send exceptions to the supervisor's strategy

### ISSUE THREAD

**Issue by connelhooley** [akkadotnet/akka.net#3200]
- Which Akka.Net version you are using
1.3.2

- On which platform you are using Akka.Net
.NET Framework 4.5.2

- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.
The following gist demonstrates the issue:
https://gist.github.com/connelhooley/14065404cb7b063c1b051429bae34bbf

Exceptions thrown by an actor do not get sent to it's supervisor's strategy when created using the ActorOfAsTestActorRef method with a supervisor ref.

**Comment by Aaronontheweb** [comment]
Yep, after looking at the reproduction this sure looks like a bug with the TestKit.

**Comment by connelhooley** [comment]
@Aaronontheweb is there a time frame for when this will be fixed? Or can you point me in the right direction for providing a fix myself? I'm not very familiar with how Akka.NET works internally so some steer on where to start would be helpful.

**Comment by Aaronontheweb** [comment]
@connelhooley I don't have a timeframe for when I'll be personally looking into it, but if you can take a stab at it we'd really appreciate it! I'd start by looking at the `TestActorRef` infrastructure itself: https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.TestKit/TestActorRef.cs

And some other helpful guidelines for contributing to Akka.NET can be found here: http://getakka.net/community/building-akka-net.html

**Comment by connelhooley** [comment]
Finally has some free time to look at this :) I think I can see why the error is happening.

We appear to his this line when the parent tries to supervise the child:

**Comment by connelhooley** [comment]
OK so I think I have a workaround and suggested fix.

If you take the following line in my gist:

``` C#
var child = ActorOfAsTestActorRef<FailingChildActor>(parent); 
````

And replace it with:

``` C#
var child = ActorOfAsTestActorRef<FailingChildActor>(parent.Ref); 
```

The code works correctly. For now I'm happy to use with this workaround in my project. A fix though could be to add additional overloads to the TestKitBase like this for example:

``` C#
public TestActorRef<TActor> ActorOfAsTestActorRef<TActor, TParentActor>(TestActorRef<TParentActor> supervisor, string name = null) where TActor : ActorBase, new() where TParentActor : ActorBase, new()
{
    return new TestActorRef<TActor>(Sys, Props.Create<TActor>(), supervisor.Ref, name);
}
```

Notice the use of `.Ref`.

Although I guess it's worrying that the Ref property is acting differently to the TestActorRef being casted to a IActorRef, because I imagine they're exposing the same InternalActorRef? I'll have a deeper look into it when I next have some spare time.
